### PR TITLE
Fixes #8091: connect-src accepts WSS

### PR DESCRIPTION
**Pull request by dLobatog**

### CODE REVIEW

**Comment by domcleal** [comment]
`#` in commit/PR title

**Comment by shlomizadok** [comment]
TestableControllerTest should also include this change.

File: config/initializers/secure_headers.rb [hunk 1]
```diff
@@ -13,7 +13,7 @@
     :enforce     => true,
     :default_src => 'self',
     :frame_src   => 'self',
-    :connect_src => 'self ws://',
```
Comment (dLobatog) [inline config/initializers/secure_headers.rb:16]: Just for the record, `ws://` is not recognized as proper connect_src syntax. `ws://*` would work too but I preferred to just specify the protocol which works too.

**Comment by domcleal** [comment]
Commit message and PR title are still missing #

**Comment by dLobatog** [comment]
Updated, sorry

**Comment by domcleal** [comment]
This does work correctly with wss:// now, but I note that ws:// doesn't work in Chrome with an unencypted websockets connection:

<pre>
[blocked] The page at 'https://foreman.example.com/compute_resources/2-localhost/vms/ea42c018-1fab-4c3c-8788-eda2351d1e82/console' was loaded over HTTPS, but ran insecure content from 'ws://foreman.example.com:5912/': this content should also be loaded over HTTPS.
</pre>

Not all websockets connections will be encrypted, e.g. if the user hasn't configured it for that, or if they're using [libvirt and SPICE](http://projects.theforeman.org/issues/7018).

**Comment by dLobatog** [comment]
I'm not sure how can we relax this setting to allow mixed content, however it's possible to load unencrypted Websockets in chrome through this



Firefox just warns about mixed content, but won't block it


And for some reason Chromium is silent about it

**Comment by domcleal** [comment]
Okay, that's not great, but we'll have to try and fix http://projects.theforeman.org/issues/7018

**Comment by domcleal** [comment]
Merged as 8553650, thanks @eLobato.

**Pull request closed without merging**